pelican/tests/test_log.py

```diff
@@ -80,3 +83,61 @@ def do_logging():
             self.assertEqual(
                 self.handler.count_logs('Another log \\d', logging.WARNING),
                 0)
+
+    def test_fatal_logger_no_quit(self):
+        # log.FatalLogger.fatal_log_level = logging.NOTSET
+        log.init(logging.DEBUG, fatal='')
+
+        for level in ('debug', 'info', 'warning', 'error', 'critical'):
+            self._log_action(level)('%s: FatalLogger.%s()' % (__name__, level))
+
+    def test_fatal_logger_warning(self):
+        # log.FatalLogger.fatal_log_level = logging.WARNING
+        log.init(logging.DEBUG, fatal='warning')
+
+        for level in ('debug', 'info'):
+            self._log_action(level)('%s: FatalLogger.%s()' % (__name__, level))
+
+        for level in ('warning', 'error', 'critical'):
+            try:
+                self._log_action(level)("FAIL")
+            except RuntimeError:
+                continue
+            else:  # should have raised RuntimeError, but didn't
+                raise AssertionError(
+                    "Fatal set to %s but failed to exit on %s()." % (
+                        logging.getLevelname(log.FatalLogger.fatal_log_level),
+                        level
+                    )
+                )
```

Comment:
That's not the proper way to test if an exception was raised :). `unittest` has `assertRaises`, this is all you need:

```python
for level in ('warning', 'error', 'critical'):
    with self.assertRaises(RuntimeError):
        self._log_action(level)("FAIL")
```

Similarly for the other cases.

Reply:
Good to  know!  Thanks, I'll make this change :)